cmd/dagger/shell.go

```diff
@@ -1472,201 +1812,176 @@ func (h *shellCallHandler) registerBuiltins() { //nolint:gocyclo
-				}
-
-				cfg, err := h.parseArgumentValues(cmd.Context(), def, fn, args)
+			Use:         shellDepsCmdName,
+			Description: "Dependencies from the module loaded in the current context",
+			GroupID:     moduleGroup.ID,
+			Args:        NoArgs,
+			Run: func(cmd *ShellCommand, _ []string) error {
+				_, err := h.GetModuleDef(nil)
 				if err != nil {
 					return err
 				}
-
-				if st == nil {
-					h.cfg = cfg
-					return nil
-				}
-
-				return cmd.Send(h.newState().WithCall(fn, cfg))
+				return cmd.Send(h.newDepsState())
 			},
 		},
 		&ShellCommand{
-			Use:     ".deps",
-			Short:   "List module dependencies",
-			GroupID: moduleGroup.ID,
-			Args:    NoArgs,
-			RunState: func(cmd *ShellCommand, args []string, st *ShellState) error {
-				def, err := h.GetModuleDef(st)
-				if err != nil {
-					return err
-				}
-
-				ctx := cmd.Context()
-
-				deps, err := def.Source.AsModule().Dependencies(ctx)
-				if err != nil {
-					return err
-				}
-
-				// Get the Name and Description of each dependency beforehand
-				// because nameShortWrapped doesn't handle errors.
-				lines := make([]string, 0, len(deps))
-				for _, dep := range deps {
-					name, err := dep.Name(ctx)
-					if err != nil {
-						return err
-					}
-					desc, err := dep.Description(ctx)
-					if err != nil {
-						return err
-					}
-					shortDesc := strings.SplitN(desc, "\n", 2)[0]
-					if shortDesc == "" {
-						shortDesc = "-"
-					}
-					lines = append(lines, name+"\x00"+shortDesc)
-				}
-
-				var doc ShellDoc
-
-				doc.Add(
-					"Module Dependencies",
-					nameShortWrapped(lines, func(line string) (string, string) {
-						s := strings.SplitN(line, "\x00", 2)
-						return s[0], s[1]
-					}),
-				)
-
-				return cmd.Println(doc.String())
+			Use:         shellStdlibCmdName,
+			Description: "Standard library functions",
+			Args:        NoArgs,
+			Run: func(cmd *ShellCommand, _ []string) error {
+				return cmd.Send(h.newStdlibState())
 			},
 		},
 		&ShellCommand{
-			Use:   ".core [function]",
-			Short: "Load a core Dagger type",
-			// On the "Additional" command group on purpose
-			GroupID: "",
+			Use:         ".core [function]",
+			Description: "Load any core Dagger type",
+			Args:        NoArgs,
 			Run: func(cmd *ShellCommand, args []string) error {
-				ctx := cmd.Context()
-
-				if len(args) == 0 {
-					var doc ShellDoc
-
-					def := h.modDef(nil)
-
-					doc.Add(
-						"Available Functions",
-						nameShortWrapped(def.GetCoreFunctions(), func(f *modFunction) (string, string) {
-							return f.CmdName(), f.Short()
-						}),
-					)
-
-					return cmd.Println(doc.String())
-				}
-
-				st, err := h.functionCall(ctx, h.newState(), args[0], args[1:])
-				if err != nil {
-					return err
-				}
-
-				return cmd.Send(st)
+				return cmd.Send(h.newCoreState())
 			},
 		},
+		cobraToShellCommand(loginCmd),
+		cobraToShellCommand(logoutCmd),
+		// TODO: Add uninstall command when available.
```

Comment:
`uninstall` is now available on `main` - if you want to add it now. `update` isn't merged yet #8839.

Reply:
Done